### PR TITLE
Add a tool to see where people blocked on a traffic signal are trying…

### PR DESCRIPTION
**Pull request by dabreegster**
… to go

Motivation: You're debugging gridlock and see lots of people waiting at a light. The demand panel shows which directions through the light are most needed, but what're the downstream dependencies where people will be blocked next?

screencast

@michaelkirk

### CODE REVIEW

File: game/src/info/mod.rs [hunk 1]
```diff
@@ -571,6 +572,15 @@ impl InfoPanel {
                             ctx, app,
                         ))),
                     )
+                } else if let Some(x) = action.strip_prefix("routes across Intersection #") {
```
Comment (dabreegster) [inline game/src/info/mod.rs:575]: This is increasingly messy, but I'm not sure of a nice workaround yet. Info panels plumb around `Details`, which have a way of mapping button clicks to opening a different info panel tab (`hyperlinks`), warping to an object (`warpers`), and jumping to a trip at some time (`time_warpers`). Maybe the refactor would be to map button clicks to callbacks that'd produce `(close the info panel, optional transition)`

File: game/src/debug/path_counter.rs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        let mut colorer = ColorNetwork::new(app);
+        // Highlight the intersection
+        colorer
+            .unzoomed
+            .push(Color::CYAN, map.get_i(i).polygon.clone());
+        colorer
+            .zoomed
+            .push(Color::CYAN.alpha(0.5), map.get_i(i).polygon.clone());
+
+        colorer.ranked_roads(cnt.clone(), &app.cs.good_to_bad_red);
```
Comment (michaelkirk) [inline game/src/debug/path_counter.rs:58]: It's nice that you have a lot of this existing code infrastructure to do something like this relatively concisely. 👏👏👏

**Review by michaelkirk — APPROVED** [review]
LGTM

File: game/src/debug/path_counter.rs (hunk 2 above)
Comment (dabreegster) [reply game/src/debug/path_counter.rs:58]: Thanks! Lots of the dataviz is still hard to interpret -- for this one, the shades of red sometimes feels less clear than discrete color buckets, and I'm not sure displaying by rank and not absolute number even makes sense. But at least the code is consolidated, so we can improve these in the future easily.

**Pull request merged**